packages/compiler-cli/src/ngtsc/hmr/src/extract_dependencies.ts

```diff
@@ -44,7 +43,15 @@ export function extractHmrLocals(
   // variables inside of functions. Note that we filter out the class name since it is always
   // defined and it saves us having to repeat this logic wherever the locals are consumed.
   const availableTopLevel = getTopLevelDeclarationNames(sourceFile);
-  return Array.from(visitor.allReads).filter((r) => r !== name && availableTopLevel.has(r));
+  let idCounter = 0;
+
+  return {
+    local: Array.from(visitor.allReads).filter((r) => r !== name && availableTopLevel.has(r)),
+    external: Array.from(visitor.namespaceReads, (n) => ({
+      moduleName: n,
+      assignedName: `ɵhmr${idCounter++}`,
+    })),
+  };
```

Comment:
I suppose this could use the `index` parameter from `Array,from` instead of maintaining a manual counter:

```suggestion
  return {
    local: Array.from(visitor.allReads).filter((r) => r !== name && availableTopLevel.has(r)),
    external: Array.from(visitor.namespaceReads, (n, index) => ({
      moduleName: n,
      assignedName: `ɵhmr${index}`,
    })),
  };
```